Assets/MixedRealityToolkit/_Core/Devices/OpenVR/GenericOpenVRController.cs

```diff
@@ -135,272 +231,245 @@ protected void UpdateControllerData(XRNodeState State, MixedRealityInteractionMa
                 TrackingState = TrackingState.NotApplicable;
             }
 
-            //Update the interaction data source
-            interactionMapping.SetPoseValue(currentControllerData);
+            lastControllerPose = currentControllerPose;
+            currentControllerPose.Position = currentControllerPosition;
+            currentControllerPose.Rotation = currentControllerRotation;
 
-            // If our value changed raise it.
-            if (interactionMapping.Changed)
+            // Raise input system events if it is enabled.
+            if (lastState != TrackingState)
             {
-                //Raise input system Event if it enabled
-                InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, currentControllerData);
+                InputSystem?.RaiseSourceTrackingStateChanged(InputSource, this, TrackingState);
             }
 
-            if (lastState != TrackingState)
+            if (TrackingState == TrackingState.Tracked && lastControllerPose != currentControllerPose)
             {
-                InputSystem?.RaiseSourceTrackingStateChanged(InputSource, this, TrackingState);
+                if (IsPositionAvailable && IsRotationAvailable)
+                {
+                    InputSystem?.RaiseSourcePoseChanged(InputSource, this, currentControllerPose);
+                }
+                else if (IsPositionAvailable && !IsRotationAvailable)
+                {
+                    InputSystem?.RaiseSourcePositionChanged(InputSource, this, currentControllerPosition);
+                }
+                else if (!IsPositionAvailable && IsRotationAvailable)
+                {
+                    InputSystem?.RaiseSourceRotationChanged(InputSource, this, currentControllerRotation);
+                }
+            }
+        }
+
+        protected void UpdatePointerData(MixedRealityInteractionMapping interactionMapping)
+        {
+            // TODO: configure an offset pointer position for each OpenVR Controller?
```

Comment:
You don't as OpenVR doesn't support spatial grip

Reply:
But we could fake it by adding a grip position to the controller and just off setting the grip by the controller position.  This might actually be beneficial.

Reply:
Agreed, but as that is not part of the spec, let's review that in beta